README.md

```diff
@@ -0,0 +1,95 @@
+# Layer0 Auth0 Proxy
+
+Sometimes, you want to put a sensitive web application behind a login wall.
+Sometimes, you don't want to write the authentication logic yourself.
+In this repository, we provide a proxy application that authenticates through Auth0 and can be easily inserted into any Terraform deployment.
+
+
+# Usage
+
+## An Example
+
+Let's discuss what a possible deployment might look like.
+
+1. A Layer0 environment in which all of the following resources will live.
+2. A sensitive application deployed to AWS.
+3. A private load balancer that sits in front of the sensitive application.
+4. The auth0-proxy application, also deployed to AWS.
+5. A public load balancer that sits in front of the auth0-proxy application.
+
+If we hand-wave away the specifics of the sensitive application (the "myapp" service in the coming example), a sample Terraform deployment of this whole system might look like this:
+
+```
+# main.tf
+
+provider "layer0" {
+  endpoint = "${var.endpoint}"
+  token = "${var.token}"
+  skip_ssl_verify = true
+}
+
+resource "layer0_environment" "demo" {
+  name = "demo"
+}
+
+resource "layer0_load_balancer" "myapp" {
+  name = "myapp"
+  environment = "${layer0_environment.demo.id}"
+  private = true
+
+  port {
+    host_port      = 80
+    container_port = 80
+    protocol       = "http"
+  }
+}
+
+resource "layer0_service" "myapp" {
+  name          = "myapp"
+  environment   = "${layer0_environment.demo.id}"
+  load_balancer = "${layer0_load_balancer.myapp.id}"
+  # and any other values that myapp needs
+}
+
+# Here's what we do in order to add the auth0-proxy:
+module "auth0" {
+  source = "github.com/quintilesims/auth0-proxy//terraform"
+  layer0_environment_id = "${layer0_environment.demo.id}"
+  auth0_domain = "SOME AUTH0 DOMAIN"
+  auth0_client_id = "AUTH0 CLIENT ID"
+  auth0_client_secret = "AUTH0 CLIENT SECRET"
+  auth0_redirect_uri = "https://${module.auth0.load_balancer_url}"
```

Comment:
I'm not sure how this is possible because it's a circular dependency?

Reply:
Yeah, it seemed weird to me as well. I actually didn't expect it to work when I threw it in, but I didn't know how else to easily get the proxy's LB URL, and terraform is apparently super smart.

Reply:
That's pretty nifty terraform can handle that. Should we consider removing this as an input option, or at least making it optional? The only use-case I can think of for not using the proxy loadbalancer url is if you have a custom domain for that particular service, which seems like a rare use-case.

Reply:
It's also worth noting we have the option of removing the load balancer resource from this module, and expect users to handle that resource themselves; the same way we expect them to have an environment resource.

Reply:
I think I'm leaning towards hiding it behind a default.